code/dataobjects/Comment.php

```diff
@@ -267,59 +267,97 @@ public function canCreate($member = null) {
 	 * @return Boolean
 	 */
 	public function canView($member = null) {
-		if(!$member) $member = Member::currentUser();
+		$member = $this->getMember($member);
 
-		// Standard mechanism for accepting permission changes from decorators
 		$extended = $this->extendedCan('canView', $member);
-		if($extended !== null) return $extended;
+		if($extended !== null) {
+			return $extended;
+		}
+
+		if(Permission::checkMember($member, 'CMS_ACCESS_CommentAdmin')) {
+			return true;
+		}
 
-		// Allow admin
-		if(Permission::checkMember($member, 'CMS_ACCESS_CommentAdmin')) return true;
+		if($parent = $this->getParent()) {
+			return $parent->canView($member)
+				&& $parent->has_extension('CommentsExtension')
```

Comment:
Non-statically invoked static methods. :) The negative margin of the backend dev world.